opal/mca/btl/uct/btl_uct_component.c

```diff
@@ -402,6 +417,32 @@ static int mca_btl_uct_component_process_uct_md (uct_md_resource_desc_t *md_desc
     return OPAL_SUCCESS;
 }
 
+#if UCT_API > UCT_VERSION(1, 7)
+static int mca_btl_uct_component_process_uct_component (uct_component_h component, char **allowed_ifaces)
+{
+    uct_component_attr_t attr;
+    ucs_status_t ucs_status;
+    int rc;
+
+    ucs_status = uct_component_query (component, &attr);
```

Comment:
need to initialize field_mask and allocate md_resources..
@hjelmn seems like you didn't try to run it?

Reply:
It seemed to run and found components but I was probably getting lucky :). Will fix.

Reply:
Out of curiosity... Why not allocate the md_resources inside uct_component_query. Calling twice seems a little silly to me.

Reply:
Fixed.

Reply:
> Out of curiosity... Why not allocate the md_resources inside uct_component_query. Calling twice seems a little silly to me.

If uct_component_query would allocate resources, would need another API to release that memory, in the existing scheme the user is responsible for memory allocation and could use something like alloca() or std::vector which do not require explicit free